bc_obps/rls/middleware/rls.py

```diff
@@ -29,18 +30,30 @@ def __call__(self, request: HttpRequest) -> HttpResponse:
         user: Optional[User] = getattr(request, 'current_user', None)
         # Note from Dylan for later:
         # when we get to actually setting roles, we'll want to still set a role here. We might need an "Unauthenticated" role to set in this case.
-        if user:
-            self._set_user_context(user)
-        else:
-            logger.info("Anonymous user detected, skipping user context setup", exc_info=True)
+        self._set_user_context(user)
         return self.get_response(request)
 
     @staticmethod
-    def _set_user_context(user: User) -> None:
+    def _set_user_guid_and_role(cursor: CursorWrapper, user: User) -> None:
+        cursor.execute('set my.guid = %s', [str(user.user_guid)])
+        # set the role based on the user's app role
+        cursor.execute('set role %s', [user.app_role.role_name])
+
+    @staticmethod
+    def _reset_user_guid_and_role(cursor: CursorWrapper) -> None:
+        cursor.execute('reset my.guid')
+        cursor.execute('reset role')
+
+    def _set_user_context(self, user: Optional[User]) -> None:
+        """
+        Sets the database session context for the given user, including their GUID and role.
+        If no user is provided, resets the context.
```

Comment:
I'd add a little bit of context here about our findings on Django reusing database connections
```py
"""
Django guarantees that each API request will be handled on the same database 
connection, but it doesn't create a new, fresh postgres session for each API request.
This means that if django is configured to use a db connection pool, context values, 
like `set role` and `set some_param`, might bleed between user requests if they're 
not properly reset by this middleware.
"""